usl_models/tests/flood_ml/mock_dataset.py

```diff
@@ -77,3 +90,48 @@ def generator():
     if batch_size:
         dataset = dataset.batch(batch_size)
     return dataset
+
+
+def mock_prediction_dataset(
+    params: model_params.FloodModelParams,
+    height: int = 100,
+    width: int = 100,
+    batch_count: int = 1,
+    batch_size: int | None = 1,
+    n: int | None = None,
+) -> tf.data.Dataset:
+    """Constructs a dataset of random mock prediction data.
+
+    Args:
+        params: Flood model params
+        height: spatial map height
+        width: spatial map width
+        batch_count: number of batches to produce
+        batch_size: size of each batch
+        n: optional number of timesteps.
+            Required for producing data for use with `model.call_n`.
+
+    Returns:
+        The TF dataset.
+    """
+    input_sig = input_signature(params, height, width, n)
+    metadata_sig = metadata_signature()
+
+    def generator():
+        """Generate random tensors."""
+        batch_size_int = 1 if batch_size is None else batch_size
+        for i in range(batch_size_int * batch_count):
+            yield flood_model.FloodModel.Input(
+                geospatial=tf.random.normal(shape=input_sig["geospatial"].shape),
+                temporal=tf.random.normal(shape=input_sig["temporal"].shape),
+                spatiotemporal=tf.random.normal(
+                    shape=input_sig["spatiotemporal"].shape
+                ),
+            ), dict(feature_chunk=str(i), rainfall=i)
+
+    dataset = tf.data.Dataset.from_generator(
+        generator=generator, output_signature=(input_sig, metadata_sig)
+    )
+    if batch_size is not None:
```

Comment:
it looks like this leads to a different behaviour if batch_size is 1 or None? Is that intentional? Might be worth documenting if that's intentional. If not, I'd just have `batch_size: int = 1` so that there aren't two ways to express the same thing.

Reply:
The difference is whether a batch dimension is added. I've noted this in the docstring.